### PR TITLE
[docker] Update cgroup mount point parsing to support alt format.

### PR DESCRIPTION
**Pull request by conorbranagan**
### What does this PR do?

Porting over from datadog-process-agent based on a support case. The new
parsing should handle cases with `cgroup` prefixes and without in the
correct way.

### Motivation

See https://trello.com/c/sS3k74hq/593-agent-cant-find-cgroup-files

### CODE REVIEW

File: pkg/util/docker/cgroup.go [hunk 1]
```diff
@@ -446,8 +446,8 @@ func parseCgroupMountPoints(r io.Reader) map[string]string {
 	scanner := bufio.NewScanner(r)
 	for scanner.Scan() {
 		mount := scanner.Text()
-		if strings.HasPrefix(mount, "cgroup ") {
-			tokens := strings.Split(mount, " ")
+		tokens := strings.Split(mount, " ")
+		if len(tokens) >= 3 && tokens[2] == "cgroup" {
```
Comment (xvello) [inline pkg/util/docker/cgroup.go]: As I had to dig into `/etc/mtab` to understand what tokens[2] means, could you please add a `checking filesystem type is cgroup` comment?

**Review by xvello — DISMISSED** [review]
Good catch! One doc request then good to merge!

**Review by xvello — DISMISSED** [review]
⛩

**Comment by xvello** [comment]
Test fails as it picks up the `/sys` mounts and not the `/host/sys`. You could force a `Datadog.SetDefault("container_cgroup_root", "/host/sys/fs/cgroup/")` at the beginning of your test

**Comment by conorbranagan** [comment]
Just pushed an update to the test. I had done a copy/paste from our repo but the defaults are different it seems. Should pass this go I hope 🤞

**Review by xvello — APPROVED** [review]
🚢

**Pull request merged**